mage/query-modules/python/elasticsearch-synchronization.md

```diff
@@ -0,0 +1,288 @@
+"edge_index_path_schema.json", {analyzer: "mem_analyzer", index_type: "edge"}) YIELD *;
+```
+to create Elasticsearch indexes.
+
+### `index_db()`
+The method is used to serialize all vertices and relationships in MemgraphDB to Elasticsearch instance. By setting the `thread_count, max_chunk_bytes, chunk_size, max_chunk_bytes and queue_size` parameters, it is possible to get a good performance spot when indexing large quantities of documents.
+
+#### **Input**
+- `node_index: str` -> The name of the **node index**. Can be used for both **streaming and parallel bulk**.
+- `edge_index: str` -> The name of the **edge index**. Can be used for both **streaming and parallel bulk**.
+- `chunk_size: int` -> The number of docs in one chunk sent to Elasticsearch (default: 500).
+- `max_chunk_bytes: int` -> The maximum size of the request in bytes (default: 100MB).
+- `raise_on_error: bool` -> Raise `BulkIndexError` containing errors (as .errors) from the execution of the last chunk when some occur. By default, it's raised.
+- `raise_on_exception: bool` -> If `False` then don’t propagate exceptions from call to bulk and just report the items that failed as failed.
+- `max_retries: int` -> Maximum number of times a document will be retried when 429 is received, set to 0 (default) for no retries on 429.
+- `initial_backoff: float` -> The number of seconds we should wait before the first retry. Any subsequent retries will be powers of `initial_backoff * 2**retry_number`
+- `max_backoff: float` -> The maximum number of seconds a retry will wait.
+- `yield_ok: float` -> If set to `False` will skip successful documents in the output.
+- `thread_count: int` -> **Size of the threadpool** to use for the bulk requests.
+- `queue_size: int` -> Size of the **task queue** between the **main thread (producing chunks to send) and the processing threads**.
+
+The method could be called in a following way:
+```
+CALL elastic_search_serialization.index_db("node_index", "edge_index", 5, 256, 104857600, True, False, 2, 2.0, 600.0, True, 2) YIELD *;
+```
+
+#### **Output**
+- `number_of_nodes: int` -> Number of indexed nodes.
+- `number_of_edges: int` -> Number of indexed edges.
+
+### `reindex()`
+**Reindex all documents** that satisfy a given query from one index to another, potentially (if target_client is specified) on a different cluster. If you don’t specify the query you will reindex all the documents.
+
+#### **Input**
+- `updatatedObjects: List[Dict[str, Any]]` -> List of all objects that **were updated and then sent as arguments to this method** with the help of the "update trigger".
+- `source_index: Union[str, List[str]])` -> Identifies **source index(or more of them)** from where documents need to be indexed.
+- `target_index: str` -> Identifies **target index** to where documents need to be indexed.
+- `query: str` -> Query written as JSON.
+- `chunk_size: int` -> Number of docs in one chunk sent to es (default: 500).
```

Comment:
sent to where? es?

Reply:
Elasticsearch? :D please write whole words

Reply:
Yes yes Elasticsearch